libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -227,17 +250,38 @@ template sendMetrics(msg: RPCMsg): untyped =
         # metrics
         libp2p_pubsub_sent_messages.inc(labelValues = [$p.peerId, t])
 
-proc sendEncoded*(p: PubSubPeer, msg: seq[byte]) {.async.} =
-  doAssert(not isNil(p), "pubsubpeer nil!")
+proc clearSendPriorityQueue(p: PubSubPeer) =
+  if p.rpcmessagequeue.sendPriorityQueue.len == 0:
+    return # fast path
 
-  if msg.len <= 0:
-    debug "empty message, skipping", p, msg = shortLog(msg)
-    return
+  while p.rpcmessagequeue.sendPriorityQueue.len > 0 and
+      p.rpcmessagequeue.sendPriorityQueue[0].finished:
+    discard p.rpcmessagequeue.sendPriorityQueue.popFirst()
 
-  if msg.len > p.maxMessageSize:
-    info "trying to send a msg too big for pubsub", maxSize=p.maxMessageSize, msgSize=msg.len
-    return
+  while p.rpcmessagequeue.sendPriorityQueue.len > 0 and
+      p.rpcmessagequeue.sendPriorityQueue[^1].finished:
+    discard p.rpcmessagequeue.sendPriorityQueue.popLast()
+
+  when defined(libp2p_expensive_metrics):
+    libp2p_gossipsub_priority_queue_size.set(labelValues = [$p.peerId])
```

Comment:
This doesnt compile when the flag is enabled

Reply:
26a37e389f3a6ed1f4e22b4934331ca47ba40960 - might be a good idea to enable these flags in CI builds